server/methods/core/vendor.js

```diff
@@ -1,20 +1,72 @@
 import { Meteor } from "meteor/meteor";
-import { check, Match } from "meteor/check";
+import { check } from "meteor/check";
 import * as Collections from "/lib/collections";
-// import { getCurrShop } from "/server/imports/fixtures/shops";
 
 Meteor.methods({
-  "shop/vendorShop": function (vendorId, shopId) {
-  	check(vendorId, String);
-    check(shopId, String);
 
-    // const shop = getCurrShop(vendorId);
-  	Collections.Accounts.update({
-  		_id: vendorId
-  	}, {
-  		$set: {
-  			shopId
-  		}
-  	});
+  "vendor/activateVendor"(memberId) {
+    check(memberId, String);
+
+    Collections.Accounts.update({
+      userId: memberId
+    }, {
+      $set: {
+        "profile.vendorDetails.0.shopActive": true,
+        "profile.vendorDetails.0._id": memberId
+      }
+    });
+  },
+  "vendor/deactivateVendor"(memberId) {
+    check(memberId, String);
+
+    Collections.Accounts.update({
+      userId: memberId
+    }, {
+      $set: {
+        "profile.vendorDetails.0.shopActive": false
+      }
+    });
+  },
+  "vendor/getVendorId"() {
+    const profile = Collections.Accounts.find({userId: Meteor.userId()}).fetch();
+    if (profile.length > 0 && profile[0].profile !== undefined && profile[0].profile.vendorDetails !== undefined) {
```

Comment:
This code in the `if` parentheses is hard to read. Break it down to variables and use `&&` to guide against error.

Reply:
Suggestion. If `profile` is an object, use `Profie` && `Profile.profile` && `Profile.profile.vendorDetails`